contractcourt/chain_arbitrator.go

```diff
@@ -1235,25 +1251,60 @@ func (c *ChainArbitrator) FindOutgoingHTLCDeadline(chanPoint wire.OutPoint,
 	// TODO(yy): can instead query the circuit map to find the exact HTLC.
 	minRefundTimeout := uint32(math.MaxInt32)
 
-	// Iterate over all active channels to find the HTLC with the matching
-	// rHash.
+	// Find the outgoing HTLC's corresponding incoming HTLC in the circuit
+	// map.
+	rHash := outgoingHTLC.RHash
+	circuit := models.CircuitKey{
+		ChanID: scid,
+		HtlcID: outgoingHTLC.HtlcIndex,
+	}
+	incomingCircuit := c.cfg.QueryIncomingCircuit(circuit)
+
+	// If there's no incoming circuit found, we will use the default
+	// deadline.
+	if incomingCircuit == nil {
```

Comment:
Q: Could we log here that this is probably a `send payment`?

Reply:
When we send payment, we actually have an incoming circuit made, that has a chanID of zero.